app/javascript/components/student/Editor.tsx

```diff
@@ -11,12 +11,13 @@ import { FileEditor, FileEditorHandle } from './editor/FileEditor'
 import { fetchJSON } from '../../utils/fetch-json'
 import { typecheck } from '../../utils/typecheck'
 import { Iteration } from '../track/IterationSummary'
+import { useIsMounted } from 'use-is-mounted'
 
 export type Submission = {
   testsStatus: SubmissionTestsStatus
   uuid: string
```

Comment:
Instead of passing back the `uuid`, add ` link` to the `links` object that refers to this, e.g.
```typescript
{ self: { href: 'string' } }
```

Reasons:

- discourages building URLs.
- allows testsubmissions to live on different domains (can track this later).
- don't need an extra URL later if you want to re-request it.

You might ask: "But we already know the test submission URL, right?"

Yes, but it might redirect. This would be the canonical URL :)

Reply:
In our websockets connection, we would need the UUID. How do you think should we deal with this?

Reply:
@SleeplessByte Merging this one now, but feel free to reply to this one.

Reply:
Do you need the uuid, or the _channel_?

It might be that the channel is the uuid, but we don't want to call it uuid then. Instead, subscription channel or whatever.